pkg/orchestrator/connectors.go

```diff
@@ -46,11 +46,15 @@ func (c *ConnectorOrchestrator) Create(
 		return nil, cerrors.Errorf("couldn't get pipeline: %w", err)
 	}
 
+	if pl.ProvisionedBy == pipeline.ProvisionTypeConfig {
```

Comment:
Nitpick: if we assume that the orchestrator exists just for API requests (which I'm fine with at this point), I think it would be better to turn the condition around and check if the entity was provisioned by the API. In other words, I think it would be better if the orchestrator wouldn't care that there is such a thing as "config provisioned entities", it should just care that it can only mutate "pipeline provisioned entities".
```suggestion
	if pl.ProvisionedBy != pipeline.ProvisionTypeAPI {
```